torch/nn/init.py

```diff
@@ -213,13 +213,20 @@ def _calculate_fan_in_and_fan_out(tensor):
         receptive_field_size = 1
         if tensor.dim() > 2:
             receptive_field_size = tensor[0][0].numel()
-        fan_in = num_input_fmaps * receptive_field_size
-        fan_out = num_output_fmaps * receptive_field_size
+        if is_depthwise is None and num_input_fmaps == 1:
+            warnings.warn(
+                'Possible depthwise conv kernel, check and set `is_depthwise` to eliminate this warning',
+                stacklevel=2)
+        if is_depthwise:
+            fan_in, fan_out = receptive_field_size, receptive_field_size
```

Comment:
Do you have a reference why this is the correct computation for depthwise convolutions?

Reply:
This is founded when I try to reproduce some tf models. No exact reference now.
Here is an intuitive illustration. As for the depthwise convolutions, the `fan_in` and `fan_out` should be the same.

Reply:
But if I have `m = torch.nn.Conv2d(4, 2, 3, groups=2)`, the idea that `fan_in` and `fan_out` should be the same is not true anymore right?
It should be `fan_in = 2*9` and `fan_out = 9`.

Reply:
@meijieru  did you had any chance to take a look at the message above?

Reply:
Sorry for late responce.
Yeah, I didn't consider group convolution case.
Now it seems that the initialization is wrong when `groups != 1`.
And we may need something else to deal with this problem.

Reply:
I have updated the issue.